Extensions/Terraform/Src/Tasks/TerraformTaskV1/src/base-terraform-command-handler.ts

```diff
@@ -53,10 +53,10 @@ export abstract class BaseTerraformCommandHandler {
             cwd: tasks.getInput("workingDirectory")
         });
 
-        let countProviders = (commandOutput.stdout.match(/provider/g) || []).length;
+        let countProviders = ["aws", "azurerm", "google"].filter(provider => commandOutput.stdout.includes(provider)).length
```

Comment:
This does not fix erroneous warnings when a sub-module inherits a provider from the root module.

https://github.com/microsoft/azure-pipelines-extensions/issues/748#issuecomment-648311976

Reply:
Ignore me... I misread the code, heh. This PR will definitely fix that issue! 👍 

The only thing I would suggest you change is fixing the typo in the unit tests as well, and if you're up to it, adding a test for cases where a single provider occurs multiple times. (i.e. like the issue I linked)